packages/repository/src/repositories/kv.repository.bridge.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as legacy from 'loopback-datasource-juggler';
+
+import {Options, DataObject} from '../common-types';
+import {Entity} from '../model';
+
+import {KeyValueRepository, KeyValueFilter} from './kv.repository';
+
+import {juggler, ensurePromise} from './legacy-juggler-bridge';
+
+/**
+ * Polyfill for Symbol.asyncIterator
+ * See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html
+ */
+// tslint:disable-next-line:no-any
+if (!(Symbol as any).asyncIterator) {
+  // tslint:disable-next-line:no-any
+  (Symbol as any).asyncIterator = Symbol.for('Symbol.asyncIterator');
+}
+
+/**
+ * An implementation of KeyValueRepository based on loopback-datasource-juggler
+ */
+export class DefaultKeyValueRepository<T extends Entity>
+  implements KeyValueRepository<T> {
+  /**
+   * A legacy KeyValueModel class
+   */
+  kvModelClass: typeof juggler.KeyValueModel;
+
+  /**
+   * Construct a KeyValueRepository with a legacy DataSource
+   * @param ds Legacy DataSource
+   */
+  constructor(
+    private entityClass: typeof Entity & {prototype: T},
+    ds: juggler.DataSource,
+  ) {
+    // KVModel class is placeholder to receive methods from KeyValueAccessObject
+    // through mixin
+    this.kvModelClass = ds.createModel<typeof juggler.KeyValueModel>(
+      '_kvModel',
```

Comment:
I wrote a comment on this line that was somehow lost :(

Consider the following scenario: an app has two key-value models called `Stats` and `ShoppingCart`, and two KV repository classes extending `DefaultKeyValueRepository`, both using the same backing datasources. The first instance will define `_kvModel` model, the second instance will re-define this model again. From what I remember from LB 3.x days, redefining model of the same name was considering as invalid usage of juggler API.

Creating a new model class on every request has negative performance impact too.

Another problem I see here is that we are not defining any properties on the backing model class. As a result, legacy juggler is not going to verify that the input data are valid model properties. AFAIK, KVAO does not implement validations yet (see [lib/kvao/set.js](https://github.com/strongloop/loopback-datasource-juggler/blob/2b8b6719a7a988c447c921dc5be678e2b52987dc/lib/kvao/set.js)), thus I guess this second point is not a blocker for this pull request.

Ideally, I'd like our `DefaultKeyValueRepository` to leverage the same mechanism that `DefaultCrudRepository` uses:
 - The backing model has the same name as the LB4 model.
 - If the datasource already has a backing model defined, then we reuse it.
 - When defining a new backing model, LB4 property definition is converted to juggler schema.

See https://github.com/strongloop/loopback-next/blob/5a6bed821fa01433c48ad437578d5a13f8a11cd8/packages/repository/src/repositories/legacy-juggler-bridge.ts#L101-L132